nautilus/gen_kube_devsuite

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+#
+# See top-level LICENSE.rst file for Copyright information
+#
+# -*- coding: utf-8 -*-
+
+"""
+This script generates a kubernetes YAML file to run the PypeIt development suite 
+"""
+
+import shutil
+import os
+import io, yaml
+
+from pkg_resources import resource_filename
+
+from IPython import embed
+
+
+def parser(options=None):
+    import argparse
+
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+                                     description='Generate a kubernetes YAML file for a Nautilus dev suite job')
+
+    parser.add_argument('name', type=str, default=None, help='Name of the job')
+    parser.add_argument('outfile', type=str, default=None, help='Name of the YAML outfile')
+    parser.add_argument('--pypeit_branch', type=str, default='develop', help='Name of the PypeIt branch to test')
+    parser.add_argument('--dev_branch', type=str, default='develop', help='Name of the PypeIt Dev-Suite branch to use')
+    parser.add_argument('--ncpu', type=int, default=6, help='Number of CPUs request')
+    parser.add_argument('--ram', type=int, default=50, help='Amount of RAM to request (Gi)')
+    #parser.add_argument('-d', '--dryrun', default=False, action='store_true',
+    #                    help='Only list the steps')
+
+    return parser.parse_args() if options is None else parser.parse_args(options)
+
+
+def main():
+
+    pargs = parser()
+
+    # Load the default
+    def_yaml_file = os.path.join(os.getenv('PYPEIT_DEV'), 'nautilus', 'kube_dev_suite.yaml')
+    with open(def_yaml_file, 'r') as stream:
+        data = yaml.safe_load(stream)
+
+    # Modify
+    data['metadata']['name'] = pargs.name
+
+    # Resources
+    requests = data['spec']['template']['spec']['containers'][0]['resources']['requests']
+    requests['cpu'] = str(pargs.ncpu)
+    requests['memory'] = f'{pargs.ram}Gi'
+
+    # Limits
+    limits = data['spec']['template']['spec']['containers'][0]['resources']['limits']
+    limits['cpu'] = str(pargs.ncpu + 1)
+    limits['memory'] = f'{2*pargs.ram}Gi'
+
+    # Args
+    my_args = 'cd PypeIt;'
+    my_args += ' cd PypeIt;'
+    my_args += ' git checkout develop;' 
+    # Telluric
```

Comment:
I think this should call `git pull --ff-only` to get an updated version of develop, otherwise it'll just use what ever was current when the docker container was built. The same goes for the PypeIt-development-suite checkout below

Reply:
added.  great catch.